### ISSUE TITLE
[java] UseArraysAsList with method call

### ISSUE THREAD

**Issue by adangel** [pmd/pmd#3867]
**Affects PMD Version:** 6.43.0

**Rule:** [UseArraysAsList](https://pmd.github.io/latest/pmd_rules_java_performance.html#usearraysaslist)

**Description:**

When calling a method on the array element and adding the result, the rule should not be triggered.
The example code on https://pmd.github.io/latest/pmd_rules_java_performance.html#usearraysaslist explictly gives this example case, but still the rule is triggered.

**Code Sample demonstrating the issue:**

```java
import java.util.ArrayList;
import java.util.List;

public class Test {
    public void foo(Integer[] ints) {
        // could just use Arrays.asList(ints)
        List<Integer> l= new ArrayList<>(100);
        for (int i=0; i< 100; i++) {
            l.add(ints[i]);
        }
        List<Integer> anotherList = new ArrayList<>();
        for (int i=0; i< 100; i++) {
            anotherList.add(ints[i].toString()); // line 13 - false positive 
        }
    }
}
```

**Expected outcome:**

PMD reports a violation at line 13, but that's wrong. That's a false positive.
